### PR TITLE
Expose MediaStream in workers

### PR DESCRIPTION
**Pull request by youennf**
***
<a href="https://pr-preview.s3.amazonaws.com/youennf/mediacapture-extensions/pull/26.html" title="Last updated on Jun 10, 2021, 2:29 PM UTC (fd03329)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-extensions/26/d89db97...youennf:fd03329.html" title="Last updated on Jun 10, 2021, 2:29 PM UTC (fd03329)">Diff</a>

### CODE REVIEW

**Comment by jan-ivar** [comment]
@youennf Why? (I'm not opposed, just would like to know where this came up).

Naively, I'd expect streams to be useful only to access older (main-thread only) APIs?

**Comment by youennf** [comment]
> Naively, I'd expect streams to be useful only to access older (main-thread only) APIs?

If we want to expose WebAudio or RTCPeerConnection APIs to workers, it is convenient to have MediaStream available there.
And it should not be a big deal to implement.

File: index.html [hunk 1]
```diff
@@ -359,7 +359,7 @@ <h2>Transferable MediaStreamTrack</h2>
     <div>
       <p>The WebIDL changes are the following:
       <pre class="idl"
->[Exposed=(Window,Worker,AudioWorklet), Transferable]
+>[Exposed=(Window,Worker), Transferable]
```
Comment (youennf) [inline index.html:362]:
```suggestion
>[Exposed=(Window,Worker,AudioWorklet), Transferable]
```

**Comment by alvestrand** [comment]
At the moment, the only purpose would be to reuse libraries of JS code that already use MediaStreams.
We don't have any objects exposed that have MediaStream in their interface.
We should probably submit this when we have the proposal to import such an interface to Worker.

**Comment by jan-ivar** [comment]
https://github.com/w3c/mediacapture-fromelement/issues/65 would be a reason to merge this.

**Review by alvestrand — APPROVED** [review]
If this is needed to unblock the OffscreenCanvas generating a stream in a worker, I'm fine with merging it now.

**Pull request merged**